addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -1961,26 +1985,36 @@ var FieldProgressBar = AbstractField.extend({
         if (isNaN($input.val())) {
             this.do_warn(_t("Wrong value entered!"), _t("Only Integer Value should be valid."));
         } else {
-            if (e.type === 'input') {
-                this._render_value($input.val());
-                if (parseFloat($input.val()) === 0) {
+            if (e.type === 'input') { // ensure what has just been typed in the input is a number
+                // returns NaN if not a number
+                var inputVal = parseFloat($input.val());
+                this._render_value(inputVal);
+                if (inputVal === 0) {
                     $input.select();
                 }
-            } else {
+            } else { // Implicit type === 'blur': we commit the value
+                var valToUpdate;
                 if (this.edit_max_value) {
-                    this.max_value = $(e.target).val();
+                    valToUpdate = $(e.target).val() || "100";
                 } else {
-                    this.value = $(e.target).val() || 0;
+                    valToUpdate = $(e.target).val() || "0";
                 }
                 var $div = $('<div>', {class: 'o_progressbar_value'});
                 this.$('.o_progressbar_value').replaceWith($div);
                 this.write_mode = false;
 
+                valToUpdate = this._parseValue(valToUpdate);
```

Comment:
this may throw exceptions

Reply:
nope, it is protected above with the do_warn